std/random.d

```diff
@@ -1934,6 +1934,177 @@ A singleton instance of the default random number generator
     assert(rnd.take(3).sum > 0);
 }
 
+// OS RND code is based on mir.random, https://github.com/libmir/mir-random/blob/master/source/mir/random/engine/package.d
+
+version (FreeBSD)
+    version = UseFileRandom;
+version (linux)
+    version = UseFileRandom;
+
+version (Windows)
+{
+    import core.sys.windows.winbase, core.sys.windows.wincrypt,
+        core.sys.windows.windows, core.sys.windows.winerror,
+        std.windows.syserror;
+
+    struct WindowsRandom
+    {
+    private:
+        ULONG_PTR _hProvider;
+
+        void initialize() @trusted
+        {
+            // https://msdn.microsoft.com/en-us/library/windows/desktop/aa379886(v=vs.85).aspx
+            // For performance reasons, we recommend that you set the pszContainer
+            // parameter to NULL and the dwFlags parameter to CRYPT_VERIFYCONTEXT
+            // in all situations where you do not require a persisted key.
+            // CRYPT_SILENT is intended for use with applications for which the UI cannot be displayed by the CSP.
+            if (!CryptAcquireContextW(&_hProvider, null, null, PROV_RSA_FULL, CRYPT_VERIFYCONTEXT | CRYPT_SILENT))
+            {
+                if (GetLastError() == NTE_BAD_KEYSET)
+                {
+                    // Attempt to create default container
+                    wenforce (CryptAcquireContextA(&_hProvider, null, null, PROV_RSA_FULL,
+                    CRYPT_NEWKEYSET | CRYPT_SILENT), "Could not acquire crypt context");
+                }
+                else
+                {
+                    wenforce (false, "Could not acquire crypt context");
+                }
+            }
+
+            popFront();
+        }
+
+    public:
+        enum bool isUniformRandom = true;
+        enum empty = false;
+
+        ulong front;
+
+        void popFront() @trusted nothrow @nogc
+        {
+            if (!CryptGenRandom(_hProvider, front.sizeof, cast(PBYTE) &front))
+                assert(0, "CryptGenRandom failed");
+        }
+
+        // TLS data is not destructed anyway right now...
+        version (none)
```

Comment:
Is this really an argument to use `version (none)`?
At least without `version (none)` the code in the destructor will get checked for errors.

Reply:
I'd have to make the `osRNDGen` return value reference counted then, otherwise the destructor would be called multiple times because of copies. And given that the value is never destructed anyway, this seems like an unnecessary complication. Do RefCounted!T ranges even work properly in all cases?

I could also remove the destructor completely. Only if phobos is used as a shared library (e.g. you load a D plugin into a C host), then you would expect these resources to be properly freed. But it doesn't seem likely destruction of static values will be implemented anytime soon...